test/test_models.py

```diff
@@ -74,12 +74,47 @@ def get_available_video_models():
 )
 
 
+# The following contains configuration parameters for all models which are used by
+# the _test_*_model methods.
+_model_params = {
+    'default_classification': {
+        'num_classes': 50,
+        'input_shape': (1, 3, 224, 224),
+    },
+    'default_segmentation': {
+        'num_classes': 10,
+        'pretrained_backbone': False,
+        'input_shape': (1, 3, 32, 32),
```

Comment:
nit: looks like we don't need to tweak `input_shape` for segmentation  and detection models, so I would leave it out for these. In fact I would even leave it out for the classification ones because we can just move

```py
input_shape = (1, 3, 299, 299) if model_name == 'inception_v3' else (1, 3, 224, 224)
``` 

in _test_classification_model

Reply:
what I meant to say  is that this:

```py
input_shape = kwargs.pop('input_shape')
```

is a little awkward so we might want to remove those from the dict. If we do, we wouldn't need to remove e.g. `input_shape = (1, 3, 32, 32)` line 105.

But again this is a nit, and perhaps I'm missing something   so feel free to leave as-is